mlpipeline/pipeline/pipeline.py

```diff
@@ -0,0 +1,135 @@
+import logging
+from ..utils import get_pipeline_message
+
+##
+L = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+##
+
+
+class Pipeline:
+    """
+    Executes components in topological order and obtains the result
+
+    ...
+
+    Methods
+    -------
+    execute(input: dict)
+        obtains result
+    """
+
+    def __init__(self, name: str, inputs: set, outputs: set, components: dict, running_order: list):
+        """
+        Parameters
+        ----------
+        name : str
+            Human readable name of the pipeline
+        inputs : set
+            input keys
+        output : set
+            output keys
+        components : dict
+            all components component_id -> component
+        running_order : list
+            component ids in topological order
+        """
+
+        self.name = name
+        self.inputs = inputs
+        self.outputs = outputs
+        self.components = components
+        self.running_order = running_order
+
+    def _verify_inputs(self, inputs: dict) -> bool:
+        """Verifies the input from cli if it matches the input expected in pipeline
+
+        Parameters
+        ----------
+        inputs : dict
+            inputs from cli
+
+        Returns
+        -------
+        bool
+            if correct
+        """
+
+        inputs_keys = set(inputs.keys())
+        if len(inputs_keys & self.inputs) != len(self.inputs):
+            return False
+
+        return True
+
+    def execute(self, inputs: dict) -> dict:
+        """Executes components one by one in topological order
+
+        Parameters
+        ----------
+        inputs : dict
+            inputs from cli
+
+        Returns
+        -------
+        dict
+            outputs
+
+        Raises
+        ------
+        RuntimeError
+            If inputs passed from cli don"t match the pipeline inputs
+        """
+
+        if not self._verify_inputs(inputs):
+            L.error("Incorrect inputs from command line, shutting down")
+            raise RuntimeError("Inputs passed from command line don't match specified inputs")
+
+        result = {}
+        result.update(inputs)
+        L.info("Starting the {}".format(self.name))
```

Comment:
Nitpick: f-strings are imho the golden standard now in Python when it's possible to use them.